test/unit/org/apache/cassandra/cql3/SimpleQueryTest.java

```
@@ -561,6 +564,36 @@ public void testSStableTimestampOrdering() throws Throwable
         assertRows(execute("SELECT * FROM %s WHERE k1=1"), row(1, 1, 2));
     }
 
+    @Test()
+    public void testInvalidCompactionOptions()
+    {
+        try {
+            createTable("CREATE TABLE %s.test (k int PRIMARY KEY, v int) WITH compaction = {" +
+                    "'class': 'LeveledCompactionStrategy', " +
+                    "'fanout_size': '90', " +
+                    "'sstable_size_in_mb': '1089'}");
+        }
+        catch (ConfigurationException e) {
+            assertTrue(e.getMessage().contains("your maxSSTableSize must be absurdly high to compute"));
+            return;
+        }
+        fail("fanout_sizeed and sstable_size_in_mb are invalid, but did not throw ConfigurationException");
+    }
+
+    @Test()
+    public void testValidCompactionOptions()
+    {
+        try {
+            createTable("CREATE TABLE %s.test (k int PRIMARY KEY, v int) WITH compaction = {" +
+                    "'class': 'LeveledCompactionStrategy', " +
+                    "'fanout_size': '10', " +
+                    "'sstable_size_in_mb': '160'}");
+        }
+        catch (Exception e) {
+            fail("fanout_sizeed and sstable_size_in_mb are valid, but threw  " + e.getClass().getName());
```

can we remove the try/catch?  its cleaner to just let it bubble up than to recatch and loose context.  It also is harder to read as I have to now understand why we are catching

---

Done